### PR TITLE
Experimenting with i386 compilation

### PR DESCRIPTION
**Pull request by arcticoder**
- added VSC workspace to gitignore
- some improvements to cpp.properties
- Dockerfile improvements: added clang
- Note clang options don't get picked up by compiler-explorer for some reason
- TODO: fix compiler error, missing library when including <cstdint>

### CODE REVIEW

**Comment by arcticoder** [comment]
Referencing https://www.youtube.com/watch?v=zBkNBP00wJE

File: Dockerfile [hunk 1]
```diff
@@ -15,6 +15,8 @@ RUN echo "*** Installing Compiler Explorer ***" \
         nodejs \
         make \
         git \
+        clang-11 \
```
Comment (madduci) [inline Dockerfile]: @arcticoder can you please remove the following? Actually the compilers are being installed in the parent image. I will try to add it, as also clang 12

File: cpp.properties [hunk 2]
```diff
@@ -20,8 +20,10 @@ compiler.clang10.exe=/usr/bin/clang++-10
 compiler.clang10.name=clang 10
 compiler.clang11.exe=/usr/bin/clang++-11
 compiler.clang11.name=clang 11
+compiler.clang11.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
 compiler.clangdefault.exe=/usr/bin/clang++-11
 compiler.clangdefault.name=clang default (11)
+compiler.clangdefault.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
```
Comment (madduci) [inline cpp.properties]: by doing so, aren't we obliging everyone to use 32 bit output? Could we make it optional? Or something like:

```
compiler.clang11x86.exe=/usr/bin/clang++-11
compiler.clang11x86.name=clang 11 x86
compiler.clang11x86.options=-std=c++1z -Wall -Wextra -Wshadow -O3 -m32 -march=i386
```

**Comment by madduci** [comment]
Hi @arcticoder 

thanks for submitting a pull request. I've just added a couple of comments. The idea is welcome, but as for the changes, I believe they are then forcing everyone who pull this image to use Clang 11 on 32 bit. A better approach would be give a separated option for it.

File: Dockerfile [hunk 3]
```diff
@@ -15,6 +15,8 @@ RUN echo "*** Installing Compiler Explorer ***" \
...
+        libstdc++6 \
```
Comment (madduci) [inline Dockerfile]: we don't need this one either?

Comment (arcticoder) [reply Dockerfile]: I can remove it, just FYI I was getting errors when it wasn't included. The container was missing the libraries.

Comment (madduci) [reply Dockerfile]: I'll add it to the parent image then!

**Pull request merged**